File: sample_1_with_java/src/main/java/com/example/sample_1_with_java/ItemVO.java
package com.example.sample_1_with_java;

abstract class ItemVO {
    public static final int TYPE_HEADER=0;
    public static final int TYPE_DATA=1;

    abstract int getType();
}

class HeaderItem extends ItemVO {
    String date;

    public HeaderItem(String date){
        this.date = date;
    }

    @Override
    int getType() {
        return ItemVO.TYPE_HEADER;
    }
}

class DataItem extends ItemVO {
    int id;
    String title;
    String content;
    boolean completed;

    public DataItem(int id, String title, String content, boolean completed){
        this.id = id;
        this.title = title;
        this.content = content;
        this.completed = completed;
    }

    @Override
    int getType() {
        return ItemVO.TYPE_DATA;
    }
}

File: app/src/main/java/com/example/student/androidlab/ItemVO.kt
package com.example.student.androidlab

/*
ItemVO
화면의 투두 항목 하나하나를 표현하기 위한 VO
- 날짜
- todo
두 데이터를 동일 타입으로 처리하기 위해 상위 만들고 하위 상속으로
 */

abstract class ItemVO {
    abstract val type: Int
    companion object {
        const val TYPE_HEADER = 0
        const val TYPE_DATA = 1
    }
}

data class HeaderItem(override val type: Int = ItemVO.TYPE_HEADER, var date: String): ItemVO()
data class DataItem(override val type: Int = ItemVO.TYPE_DATA,
                    var id: Int,
                    var title: String,
                    var content: String,
                    var completed: Boolean = false): ItemVO()
